tm2/pkg/errors/errors.go

```diff
@@ -158,10 +158,11 @@ func (err *cmnError) Format(s fmt.State, verb rune) {
 		// Write stack trace.
 		if err.stacktrace != nil {
 			s.Write([]byte("Stack Trace:\n"))
-			for i, pc := range err.stacktrace {
-				fnc := runtime.FuncForPC(pc)
-				file, line := fnc.FileLine(pc)
-				fmt.Fprintf(s, " %4d  %s:%d\n", i, file, line)
+			frames := runtime.CallersFrames(err.stacktrace)
+			i := 0
+			for frame, _ := frames.Next(); frame.PC != 0; frame, _ = frames.Next() {
```

Comment:
The `Frames` loop example in the stdlib  looks more readable, shouldn't you use it ?

```go
        frames := runtime.CallersFrames(pc)

		// Loop to get frames.
		// A fixed number of PCs can expand to an indefinite number of Frames.
		for {
			frame, more := frames.Next()

            // Process this frame.
			// ....

			// Check whether there are more frames to process after this one.
			if !more {
				break
			}
		}
```

Link https://pkg.go.dev/runtime#Frames

Reply:
Didn't see that, thank you, definitely better. (Though I do think it would have been better if they made the API the same as bufio.Scanner...)